scripts/generate-javadoc.sh

```diff
@@ -16,7 +16,12 @@ mkdir_p "$ARCHIVE_DIR"
 function generate_javadoc_core() {
 	declare release=$1
 	# First we need to get the built javadocs.
-	wget --no-verbose "https://repo.jenkins-ci.org/releases/org/jenkins-ci/main/jenkins-core/${release}/jenkins-core-${release}-javadoc.jar"
+	if [ -n ${ARTIFACT_CACHING_PROXY_ORIGIN-} ]
```

Comment:
Testing it again, it was always `true` indeed. Shellcheck hinted where I made a mistake: https://www.shellcheck.net/wiki/SC2070

> `-n` doesn't work with unquoted arguments. Quote or use `[[ ]]`.

Fix:
```suggestion
	if [[ -n ${ARTIFACT_CACHING_PROXY_ORIGIN-} ]]
```

Reply:
I've picked the solution I saw elsewhere, I can always switch to the double quote one if you think it's better.

Reply:
Double quotes added.